app/controllers/auth_controller.rb

```diff
@@ -2,16 +2,36 @@ class AuthController < ApplicationController
   skip_before_action :authenticate_user!, only: [:oauth_callback]
 
   def oauth_callback
-    if team_member?
-      user = User.find_or_create_by(email: auth_email)
-      sign_in(user)
-      flash[:success] = "You successfully signed in"
-      redirect_to root_path
+    if ok_to_bypass_authentication?
+      create_user(developer_email)
+    else
+      if team_member?
+        create_user(auth_email)
+      end
```

Comment:
Not a blocker for this pull request, but there should probably be an `else` here.